lib/app_profiler/yarn/command.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module AppProfiler
+  module Yarn
+    module Command
+      class YarnError < StandardError; end
+
+      mattr_accessor :yarn_setup, default: false
+
+      def yarn(command)
+        setup_yarn unless yarn_setup
+        exec("yarn #{command}") do
+          raise YarnError, "Failed to run #{command}."
+        end
+      end
+
+      def setup_yarn
+        ensure_yarn_installed
+        yarn("init --yes") unless package_json_exists?
+      end
+
+      private
+
+      def ensure_yarn_installed
+        exec("which yarn > /dev/null") do
+          raise(
+            YarnError,
+            <<~MSG.squish
+              `yarn` command not found.
+              Please install `yarn` or make it available in PATH.
+            MSG
+          )
+        end
+        self.yarn_setup = true
+      end
+
+      def package_json_exists?
+        AppProfiler.root.join("package.json").exist?
+      end
+
+      def exec(command)
+        system(command).tap do |return_code|
+          yield unless return_code
+        end
```

Comment:
Similarly, this seems needlessly dangerous. Instead of accepting any command and passing it to `system`, we should allow list what we want to run or just write individual methods with no variability

Reply:
It is a private method and the only consumer is `yarn`. I can inline it if that would make sense to you?